src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -252,5 +267,67 @@ public void Dispose()
         {
             _deadlineTimer?.Dispose();
         }
+
+        internal string GetRequestGrpcEncoding()
+        {
+            if (HttpContext.Request.Headers.TryGetValue(GrpcProtocolConstants.MessageEncodingHeader, out var values))
+            {
+                return values;
+            }
+
+            return null;
+        }
+
+        internal bool IsEncodingInRequestAcceptEncoding(string encoding)
+        {
+            if (HttpContext.Request.Headers.TryGetValue(GrpcProtocolConstants.MessageAcceptEncodingHeader, out var values))
```

Comment:
> Currently however, compression levels are only supported at the server side, which is aware of the client's capabilities through the incoming Message-Accept-Encoding header.

I'm reading the specs and the emails again and I'd like to confirm that "Message-Accept-Encoding" refers to `grpc-accept-encoding` instead of the `accept-encoding` defined by HTTP

Reply:
Yes. MessageAcceptEncoding is what it is called in the gRPC HTTP2 spec.